platforms/allwinner-d1/src/dmac/mod.rs

```diff
@@ -13,55 +13,162 @@ use d1_pac::{
 };
 
 use crate::ccu::Ccu;
+use kernel::maitake::sync::WaitCell;
 
 use self::descriptor::Descriptor;
 
 pub mod descriptor;
 
 pub struct Dmac {
-    pub dmac: DMAC, // TODO: not this
-    pub channels: [Channel; Self::CHANNEL_COUNT as usize],
+    state: &'static DmacState,
+    dmac: DMAC,
 }
 
+pub struct Channel {
+    idx: u8,
+    channel: &'static ChannelState,
+}
+
+struct DmacState {
+    used_channels: AtomicU8,
+    channels: [ChannelState; Dmac::CHANNEL_COUNT as usize],
+}
+
+struct ChannelState {
+    waker: WaitCell,
+    state: AtomicU8,
+}
+
+static DMAC_STATE: DmacState = {
+    // This `const` is used as a static initializer, so clippy is wrong here...
```

Comment:
I think we have a few of these, if we want a blanket allow at the top of the project, that's cool with me. It might be across multiple crates sadly tho.

Reply:
i think there might be a clippy config file or something that we could use to disable the lint globally, without having to stick an allow attribute in every crate? it's definitely Not My Favorite Clippy Lint...